pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -244,24 +251,34 @@ func computeActiveCondition(ctx context.Context, pa *pav1alpha1.PodAutoscaler, p
 
 	switch {
 	case pc.want == 0:
-		if pa.Status.IsActivating() {
+		if pa.Status.IsActivating() && minReady > 0 {
 			// We only ever scale to zero while activating if we fail to activate within the progress deadline.
 			pa.Status.MarkInactive("TimedOut", "The target could not be activated.")
 		} else {
 			pa.Status.MarkInactive("NoTraffic", "The target is not receiving traffic.")
 		}
+		// This happens in the initialScale 0 case. After SKS has been setup, we override want -1
+		// with 0 when PA's desired scale is not set yet.
+		if pc.ready >= minReady {
+			pa.Status.MarkScaleTargetInitialized()
+		}
 
 	case pc.ready < minReady:
 		if pc.want > 0 || !pa.Status.IsInactive() {
 			pa.Status.MarkActivating(
 				"Queued", "Requests to the target are being buffered as resources are provisioned.")
+		} else {
+			// Need to set a condition because otherwise we will get NewObservedGenFailure because the reconciler
+			// does not set any condition during reconciliation of a new generation
+			pa.Status.MarkInactive("NoTraffic", "The target is not receiving traffic.")
 		}
 
 	case pc.ready >= minReady:
-		if pc.want > 0 || !pa.Status.IsInactive() {
-			pa.Status.MarkScaleTargetInitialized()
-			// SKS should already be active.
+		// In the initialScale > 0 case, SKS should already be active. However, when initialScale
```

Comment:
Same here: `if pc.want == 0` which is what happens when we set IS=0.

Not sure that SKS is important here in the comment.

Reply:
I was trying to say even after SKS has come up, we can still get pc.want = -1 here because we only override -1 with 0 once, before we MarkScaleTargetInitialized, but pc.want can remain -1  for a few seconds